utility/importer/brave_importer.cc

```diff
@@ -0,0 +1,373 @@
+                                     in_toolbar,
+                                     bookmark_folders_dict,
+                                     bookmarks_dict,
+                                     bookmark_order_dict,
+                                     bookmarks);
+      } else {
+        // Add ImportedBookmarkEntry for empty folder.
+         ImportedBookmarkEntry imported_bookmark_folder;
+        imported_bookmark_folder.is_folder = true;
+        imported_bookmark_folder.in_toolbar = in_toolbar;
+        imported_bookmark_folder.url = GURL();
+        imported_bookmark_folder.path = path;
+        imported_bookmark_folder.title = base::UTF8ToUTF16(title);
+        // Brave doesn't specify a creation time for the folder.
+        imported_bookmark_folder.creation_time = base::Time::Now();
+        bookmarks->push_back(imported_bookmark_folder);
+      }
+    } else if (type == "bookmark") {
+      base::Value* bookmark =
+        bookmarks_dict->FindKeyOfType(key, base::Value::Type::DICTIONARY);
+      auto& title =
+        bookmark->FindKeyOfType("title", base::Value::Type::STRING)->GetString();
+      auto& location =
+        bookmark->FindKeyOfType("location", base::Value::Type::STRING)->GetString();
+
+      ImportedBookmarkEntry imported_bookmark;
+      imported_bookmark.is_folder = false;
+      imported_bookmark.in_toolbar = in_toolbar;
+      imported_bookmark.url = GURL(location);
+      imported_bookmark.path = path;
+      imported_bookmark.title = base::UTF8ToUTF16(title);
+      // Brave doesn't specify a creation time for the bookmark.
+      imported_bookmark.creation_time = base::Time::Now();
+      bookmarks->push_back(imported_bookmark);
+    }
+  }
+}
+
+void BraveImporter::ImportBookmarks() {
+  std::vector<ImportedBookmarkEntry> bookmarks;
+  ParseBookmarks(&bookmarks);
+
+  if (!bookmarks.empty() && !cancelled()) {
+    const base::string16& first_folder_name =
+      bridge_->GetLocalizedString(IDS_BOOKMARK_GROUP_FROM_BRAVE);
+    bridge_->AddBookmarks(bookmarks, first_folder_name);
+  }
+}
+
+void BraveImporter::ImportPasswords() {
+#if !defined(USE_X11)
+  base::FilePath passwords_path =
+    source_path_.Append(
+      base::FilePath::StringType(FILE_PATH_LITERAL("Login Data")));
+
+  password_manager::LoginDatabase database(passwords_path);
+  if (!database.Init()) {
+    LOG(ERROR) << "LoginDatabase Init() failed";
+    return;
+  }
+
+  std::vector<std::unique_ptr<autofill::PasswordForm>> forms;
+  bool success = database.GetAutofillableLogins(&forms);
+  if (success) {
+    for (size_t i = 0; i < forms.size(); ++i) {
+      bridge_->SetPasswordForm(*forms[i].get());
+    }
+  }
+  std::vector<std::unique_ptr<autofill::PasswordForm>> blacklist;
+  success = database.GetBlacklistLogins(&blacklist);
+  if (success) {
+    for (size_t i = 0; i < blacklist.size(); ++i) {
+      bridge_->SetPasswordForm(*blacklist[i].get());
+    }
+  }
+#else
+  base::FilePath prefs_path =
```

Comment:
actually can we pass as argument? since it is the only deviation compared to `ChromeImporter:: ImportPasswords` so that we don't have to override it